internal/pkg/agent/application/coordinator/coordinator.go

```diff
@@ -328,7 +328,7 @@ func (c *Coordinator) runner(ctx context.Context) error {
 
 	varsWatcher := c.varsMgr
 	varsRun := make(chan bool)
-	varsErrCh := make(chan error)
+	varsErrCh := make(chan error, 1)
```

Comment:
Can you explain to me why this is needed? The `select/case` reads from all these channels, so why would an unbuffered channel be needed here?

Reply:
All the places where the unbuffered channels were changed to buffered are the channels that were blocking on shutdown. So it was either something was writing to that channel that nothing was reading from (exited read loop before the writers were fully stopped) or reading from the channel that nothing was writing to.
This fix is not ideal, I understand. Might have to fix more code in order to avoid buffered channels then.

Reply:
we should aim on fixing the real issue not the symptom. 
however if we agree that this is sufficient for now add a TODO there at least and create an issue. otherwise we're creating technical debt which we will probably never deal with